### ISSUE TITLE
Slack Channel Migration to CNCF Workspace

### ISSUE THREAD

**Issue by zijianjoy** [kubeflow/community#688]
As part of [Kubeflow Onboarding to CNCF action item](https://github.com/cncf/sandbox/issues/196), we are being asked to migrate Slack channels to CNCF workspace:

> Slack: Are your slack channels migrated to the Kubernetes or CNCF Slack? (see https://slack.com/help/articles/217872578-Import-data-from-one-Slack-workspace-to-another for more details)

CNCF Slack is under Paid Slack tier, which allows chat history to be preserved. We also obtain other benefits as Paid Slack workspace. And we can establish stronger connection with other CNCF community members in this CNCF Slack workspace.

This is a discussion ticket for our decision about this item. In order to process with the CNCF transition, please comment on the questions we need to answer as below:

1. Should we migrate slack data to CNCF as an one-time action? https://slack.com/help/articles/201658943-Export-your-workspace-data
2. Should we invite people to join CNCF Slack only? Or Kubeflow Slack only?

After discussion, we will consult Kubeflow Steering Committee for the final decision.

@james-jwu @jbottum @theadactyl 

cc @terrytangyuan @akgraner @andreyvelich @thesuperzapper

**Comment by johnugeorge** [comment]
There was a similar discussion before. 

I feel that we can make a one time switch to CNCF with data and invite them to CNCF slack post switch.

**Comment by rareddy** [comment]
We need to establish a presence on the CNCF Slack because users discovering us through CNCF are likely to seek support there initially. It's essential that we reach a consensus on this matter.

Now, the question remains whether to maintain our presence on the "Kubeflow" Slack. In my view, at least for the time being, we should continue doing so. We can gradually transition the existing user base to CNCF channels. Once we observe a significant shift in traffic towards CNCF, we can consider closing out the Kubeflow Slack. It's worth noting that our current subscription is free, and we're uncertain about the duration of this complimentary access. To avoid any potential issues, we should be proactive and not find ourselves without access if Slack discontinues the free subscription unexpectedly.

**Comment by rareddy** [comment]
https://github.com/kubeflow/community/issues/615

**Comment by tarilabs** [comment]
As relatively new to this community, I think the reasonings of keeping both in the KF community meetings sounded very valid.

To start promoting the CNCF slack maybe something along the lines of:

> Join us also on the CNCF Slack!
> We recommend you join these channels:
> - https://cloud-native.slack.com/archives/{channel_a}
> - https://cloud-native.slack.com/archives/{channel_b}
> 
> (to get invited to the CNCF Slack, follow instruction [here](#))

This as a way to at least mitigate on the limitation of channels-auto-join?
Just an idea.

**Comment by andreyvelich** [comment]
Thanks for creating this @zijianjoy!
As I said on the Kubeflow Community Call, I would like to get feedback from other communities who migrated to CNCF Slack.
@terrytangyuan I noticed you left a few comments here: https://github.com/cncf/toc/issues/1139#issuecomment-1792574906

@terrytangyuan How do you manage communications with Argo users in the CNCF Slack ? 
Since you can't post messages in the #announcement channel and when users join CNCF Slack, they have to manually search for Argo channels.

Also, do we have anyone from other communities to provide feedback (e.g. KNative, Helm)? 

Also, would it be possible to configure Slack invitation which automatically adds Kubeflow channel for users in CNCF Slack?

**Comment by terrytangyuan** [comment]
> How do you manage communications with Argo users in the CNCF Slack ?

There's #argo-announcements channel. It's ok for them to search for Argo related channels since we'll know how many users are actually interested in a particular subproject. We also have a page that lists the main channels so that users can find them easily https://argoproj.github.io/community/join-slack 

> Also, would it be possible to configure Slack invitation which automatically adds Kubeflow channel for users in CNCF Slack?

I don't think so. IMO tt's important for them to search and join any channels they are interested in instead of forcing them to join all subproject channels and later leave/mute all of them.

**Comment by terrytangyuan** [comment]
@zijianjoy 

> Should we migrate slack data to CNCF as an one-time action? https://slack.com/help/articles/201658943-Export-your-workspace-data

I don't know how useful it is. 

> Should we invite people to join CNCF Slack only? Or Kubeflow Slack only?

I think we should only invite new people to join CNCF Slack and follow a similar process to what we did for Argo project. See https://github.com/cncf/toc/issues/1139#issuecomment-1793054634

**Comment by zijianjoy** [comment]
@terrytangyuan 
> > Should we migrate slack data to CNCF as an one-time action? https://slack.com/help/articles/201658943-Export-your-workspace-data
> 
> I don't know how useful it is.
> 

This is for porting existing kubeflow channels and recent messages to CNCF.

**Comment by terrytangyuan** [comment]
>This is for porting existing kubeflow channels and recent messages to CNCF.

IMO I don't find those messages useful. Users can always stay in the Kubeflow Slack to see what was there. I'd prefer looking forward and it would be great to hear from others too.

**Comment by zijianjoy** [comment]
> > This is for porting existing kubeflow channels and recent messages to CNCF.
> 
> IMO I don't find those messages useful. Users can always stay in the Kubeflow Slack to see what was there. I'd prefer looking forward and it would be great to hear from others too.

In Kubeflow Community Meeting, Kubeflow WG leads wish to have the same channels available in CNCF, so they can continue the conversation and enable smooth slack channel discovery for the community.

**Comment by terrytangyuan** [comment]
Sure, it doesn't hurt. Although the channel names will have prefix "kubeflow-" so I am not sure if we can import the chat history correctly. There are also channels names that collide with existing CNCF channels.

**Comment by thesuperzapper** [comment]
My current idea is that we should:

1. Leave the existing Kubeflow slack in place
2. Make a small number of channels in the CNCF slack for the official components (but maybe even leave it to the already-present `#kubeflow` channel)
3. Publicise that these channels exist [on the website,](https://www.kubeflow.org/docs/about/community/#kubeflow-slack) alongside the existing Slack link

Then, if we see a significant adoption of the new CNCF channels, we can specifically designate the "Kubeflow Slack" for development/working-groups and use the "CNCF Slack" for users. But otherwise, we just maintain the presence in both places.

I suggest this because, as I was saying in https://github.com/cncf/toc/issues/1139#issuecomment-1854934204, we have many active users across 38 active channels in the Kubeflow slack (+ about 75% of the activity occurs in private messages), with more users joining daily (about 150 in the last few weeks over Christmas) so I want to avoid killing that momentum until we see some benefit.

**Comment by terrytangyuan** [comment]
> Then, if we see a significant adoption of the new CNCF channels, we can specifically designate the "Kubeflow Slack" for development/working-groups and use the "CNCF Slack" for users. But otherwise, we just maintain the presence in both places.

If we see significant adoption, why would we still use the Kubeflow Slack? For contributors/maintainers, quite few of us have already been communicating in CNCF Slack for years since that's where the majority of collaboration among CNCF projects happen. 

> about 75% of the activity occurs in private messages

You'll see a lot more activities in CNCF channel, especially private messages since the chat history is there and you get to talk with more maintainers from other CNCF projects.

**Comment by terrytangyuan** [comment]
As announced in a previous community meeting, the Kubeflow Steering Committee has made the decision to transition to CNCF Slack going forward.

**Comment by thesuperzapper** [comment]
I was under the impression from the meeting that we would be keeping both Slacks for the time being, in a cross over similar to what I proposed in https://github.com/kubeflow/community/issues/688#issuecomment-1896962143.

Also, we have effectively been following that plan for months (we already have channels on the CNCF Slack), we just need to advertise the existence of those CNCF Slack channels.

PS: this is pretty much exactly the situation that the Kubernetes slack is currently in, where they have a presence on both the official CNCF one in addition their own Slack, because migrating such a large number of channels and people is very hard.

---

There was also some discussion in that meeting about wanting the ability to do announcements (e.g. `@here`) and that the existing slack is actually still growing very quickly (including in channels and people).

For reference, we added about 200 people in the last few weeks and (we now have 10,220 total), we also added a bunch of other new channels for our new projects/groups, so we are at 40+ active channels.

**Comment by terrytangyuan** [comment]
Yes, the Kubeflow Slack workspace will still be there but we will gradually move towards CNCF Slack. We are still working on the logistics. Please stay tuned for the announcement.
